contracts/staking/StakingInfo.sol

```diff
@@ -220,14 +232,16 @@ contract StakingInfo {
         address newSigner,
         bytes memory signerPubkey
     ) public onlyStakeManager {
-        emit SignerChange(validatorId, oldSigner, newSigner, signerPubkey);
+        validatorNonce[validatorId] = validatorNonce[validatorId].add(1);
+        emit SignerChange(validatorId, validatorNonce[validatorId], oldSigner, newSigner, signerPubkey);
     }
 
     function logReStaked(uint256 validatorId, uint256 amount, uint256 total)
         public
         onlyStakeManager
-    {
-        emit ReStaked(validatorId, amount, total);
+    {   
+        validatorNonce[validatorId] = validatorNonce[validatorId].add(1);
+        emit ReStaked(validatorId, validatorNonce[validatorId], amount, total);
```

Comment:
Only the following events need to emit with `nonce` (not any other event):

`Staked`, `StakeUpdate` , `SignerChange` , `UnstakeInit`